src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -368,25 +386,38 @@ private void subscribe(String topic) {
         try {
             SubAckPacket subAckPacket = client.subscribe(subscribePacket)
                     .get(MQTT_OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
-            if (subAckPacket.getReasonCodes().stream().allMatch(this::subscriptionIsSuccessful)) {
-                synchronized (subscriptionsLock) {
-                    subscribedLocalMqttTopics.add(topic);
-                }
-                LOGGER.atDebug()
+            if (subAckPacket.getReasonCodes().stream().allMatch(this::retrySubscribe)) {
+                // received a retryable reason code
+                LOGGER.atError()
                         .kv(LOG_KEY_REASON_CODES, subAckPacket.getReasonCodes())
                         .kv(LOG_KEY_REASON, subAckPacket.getReasonString())
                         .kv(LOG_KEY_TOPIC, topic)
-                        .log("Successfully subscribed to topic");
+                        .log("Failed to subscribe to topic");
```

Comment:
also mention that we're retrying

Reply:
Don't log anything here. But make the exception message logged by the retry utils useful enough to know what's going on. Include the topic and the reason cdoe.

Reply:
You don't change retryutils. The error message in the exception is what will be logged, so change that.